odoo/addons/test_new_api/tests/test_domain.py

```diff
@@ -39,3 +41,71 @@ def test_00_test_bool_undefined(self):
+        self.assertListEqual(EmptyChar.search([('name', '!=', 'name')]).mapped('name'), ['', False])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', 'name')]).mapped('name'), ['name'])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', 'name')]).mapped('name'), ['', False])
+
+        self.assertListEqual(EmptyChar.search([('name', '=', '')]).mapped('name'), [''])
+        self.assertListEqual(EmptyChar.search([('name', '!=', '')]).mapped('name'), ['name'])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', '')]).mapped('name'), ['name', '', False])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', '')]).mapped('name'), [False])
+
+        self.assertListEqual(EmptyChar.search([('name', '=', False)]).mapped('name'), [False])
+        self.assertListEqual(EmptyChar.search([('name', '!=', False)]).mapped('name'), ['name', ''])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', False)]).mapped('name'), ['name', '', False])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', False)]).mapped('name'), [False])
+
+        values = ['name', '', False]
+        for length in range(len(values) + 1):
+            for subset in combinations(values, length):
+                sublist = list(subset)
+                self.assertListEqual(EmptyChar.search([('name', 'in', sublist)]).mapped('name'), sublist)
+                sublist_remained = [v for v in values if v not in subset]
+                self.assertListEqual(EmptyChar.search([('name', 'not in', sublist)]).mapped('name'), sublist_remained)
+
+    def test_empty_translation(self):
+        records_en = self.env['test_new_api.indexed_translation'].with_context(lang='en_US').create([
+            {'name': 'English'},
+            {'name': 'English'},
+            {'name': 'English'},
+        ])
+        self.env['res.lang']._activate_lang('fr_FR')
+        records_fr = records_en.with_context(lang='fr_FR')
+        records_fr[0].name = 'name'
+        records_fr[1].name = ''
+        records_fr[2].name = False
+
+        self.assertListEqual(records_fr.search([('name', '=', 'name')]).mapped('name'), ['name'])
+        self.assertListEqual(records_fr.search([('name', '!=', 'name')]).mapped('name'), ['', False])
+        self.assertListEqual(records_fr.search([('name', 'ilike', 'name')]).mapped('name'), ['name'])
+        self.assertListEqual(records_fr.search([('name', 'not ilike', 'name')]).mapped('name'), ['', False])
+
+        self.assertListEqual(records_fr.search([('name', '=', '')]).mapped('name'), [''])
+        self.assertListEqual(records_fr.search([('name', '!=', '')]).mapped('name'), ['name'])
+        # False and '' are treated as different value when (need_wildcard and not right)
+        self.assertListEqual(records_fr.search([('name', 'ilike', '')]).mapped('name'), ['name', '', False])
+        self.assertListEqual(records_fr.search([('name', 'not ilike', '')]).mapped('name'), [False])
```

Comment:
@rco-odoo could you check this behavior for non-translated field?
Intuitively, I think `False` record should be returned in either `ilike` or `not like`.